### ISSUE TITLE
odd interactions or potential bug  [1.16.4] Causing world gen to halt for 5 to 10 mins before starting to generate, not sure of actual cause

### ISSUE THREAD

**Issue by SDUBZ** [ldtteam/minecolonies#6112]
### Prerequisites 

- [x] I am running the latest **alpha** version of MineColonies and Structurize for my Minecraft version.
- [x] I checked the MineColonies/Structurize [wiki](https://wiki.minecolonies.ldtteam.com/) and made sure my issue is not covered there.
- [x ] I made sure that **this issue is not a duplicate of any existing issue.**

### Context 

- Minecraft Version:1.16.4
forge: 1.16.4-35.0.17
- MineColonies Version: minecolonies-0.13.452-ALPHA-universal
- Structurize Version: structurize-0.13.96-ALPHA-universal

## Problem description
Causes a big hang time only when i have minecolonies added
tried disabling many others and got same thing every time i tried , creating new worlds or reloading instance would get same big hang time before generating

### Notes or related things
this not happen with structurize itself though

logs:
done over time with multiple testing's to try eliminate the cause

a earlier debug:
https://gist.github.com/SDUBZ/0d57d0737b58cd1d29e110fcf28f6349

with more suspected mods disabled:
https://gist.github.com/SDUBZ/6dd776c657d9dcbdebbef88ec6059c63

Latest debug without minecolonies:
https://gist.github.com/SDUBZ/39e12dc9e279d6b768bae0b192e9007a

list of mods
[mods.txt](https://github.com/ldtteam/minecolonies/files/5565736/mods.txt)

---
## Viewers

* Add a thumbs-up to the issue to vote for it. This helps the request become more visible to the team and doesn't clutter the comments.
* Add a comment if you have any insights or background information that isn't already part of the conversation.

**Comment by Raycoms** [comment]
Not sure I understand your problem description?

**Comment by Nightenom** [comment]
btw from your debug.log
```
[19Nov2020 17:03:30.302] [Render thread/FATAL] [net.minecraft.util.concurrent.ThreadTaskExecutor/]: Error executing task on Client
java.lang.IndexOutOfBoundsException: readerIndex(5) + length(4) exceeds writerIndex(5): UnpooledByteBufAllocator$InstrumentedUnpooledUnsafeHeapByteBuf(ridx: 5, widx: 5, cap: 256)
	at io.netty.buffer.AbstractByteBuf.checkReadableBytes0(AbstractByteBuf.java:1405) ~[netty-all-4.1.25.Final.jar:4.1.25.Final]
	at io.netty.buffer.AbstractByteBuf.readInt(AbstractByteBuf.java:786) ~[netty-all-4.1.25.Final.jar:4.1.25.Final]
	at net.minecraft.network.PacketBuffer.readInt(PacketBuffer.java:795) ~[?:?]
	at com.supermartijn642.movingelevators.packets.ElevatorMovementPacket.decode(ElevatorMovementPacket.java:42) ~[movingelevators:?]
	at com.supermartijn642.movingelevators.packets.ElevatorMovementPacket.<init>(ElevatorMovementPacket.java:30) ~[movingelevators:?]
	at net.minecraftforge.fml.network.simple.IndexedMessageCodec.lambda$tryDecode$0(IndexedMessageCodec.java:120) ~[forge:?]
	at java.util.Optional.map(Optional.java:215) ~[?:1.8.0_242]
	at net.minecraftforge.fml.network.simple.IndexedMessageCodec.tryDecode(IndexedMessageCodec.java:120) ~[forge:?]
	at net.minecraftforge.fml.network.simple.IndexedMessageCodec.consume(IndexedMessageCodec.java:162) ~[forge:?]
	at net.minecraftforge.fml.network.simple.SimpleChannel.networkEventListener(SimpleChannel.java:80) ~[forge:?]
	at net.minecraftforge.eventbus.EventBus.doCastFilter(EventBus.java:247) ~[eventbus-3.0.5-service.jar:?]
	at net.minecraftforge.eventbus.EventBus.lambda$addListener$11(EventBus.java:239) ~[eventbus-3.0.5-service.jar:?]
	at net.minecraftforge.eventbus.EventBus.post(EventBus.java:297) ~[eventbus-3.0.5-service.jar:?]
	at net.minecraftforge.fml.network.NetworkInstance.dispatch(NetworkInstance.java:84) ~[forge:?]
	at net.minecraftforge.fml.network.NetworkHooks.lambda$onCustomPayload$1(NetworkHooks.java:88) ~[forge:?]
	at java.util.Optional.map(Optional.java:215) ~[?:1.8.0_242]
	at net.minecraftforge.fml.network.NetworkHooks.onCustomPayload(NetworkHooks.java:88) ~[forge:?]
	at net.minecraft.client.network.play.ClientPlayNetHandler.func_147240_a(ClientPlayNetHandler.java:1904) ~[?:?]
	at net.minecraft.network.play.server.SCustomPayloadPlayPacket.func_148833_a(SCustomPayloadPlayPacket.java:59) ~[?:?]
	at net.minecraft.network.play.server.SCustomPayloadPlayPacket.func_148833_a(SCustomPayloadPlayPacket.java:11) ~[?:?]
	at net.minecraft.network.PacketThreadUtil.func_225383_a(SourceFile:21) ~[?:?]
	at net.minecraft.util.concurrent.ThreadTaskExecutor.func_213166_h(SourceFile:144) ~[?:?]
	at net.minecraft.util.concurrent.RecursiveEventLoop.func_213166_h(SourceFile:23) ~[?:?]
	at net.minecraft.util.concurrent.ThreadTaskExecutor.func_213168_p(SourceFile:118) ~[?:?]
	at net.minecraft.util.concurrent.ThreadTaskExecutor.func_213160_bf(SourceFile:103) ~[?:?]
	at net.minecraft.client.Minecraft.func_195542_b(Minecraft.java:947) ~[?:?]
	at net.minecraft.client.Minecraft.func_99999_d(Minecraft.java:607) ~[?:?]
	at net.minecraft.client.main.Main.main(Main.java:184) ~[1.16.4.jar:?]
	at sun.reflect.NativeMethodAccessorImpl.invoke0(Native Method) ~[?:1.8.0_242]
	at sun.reflect.NativeMethodAccessorImpl.invoke(NativeMethodAccessorImpl.java:62) ~[?:1.8.0_242]
	at sun.reflect.DelegatingMethodAccessorImpl.invoke(DelegatingMethodAccessorImpl.java:43) ~[?:1.8.0_242]
	at java.lang.reflect.Method.invoke(Method.java:498) ~[?:1.8.0_242]
	at net.minecraftforge.fml.loading.FMLClientLaunchProvider.lambda$launchService$0(FMLClientLaunchProvider.java:51) ~[forge-1.16.4-35.0.18.jar:35.0]
	at cpw.mods.modlauncher.LaunchServiceHandlerDecorator.launch(LaunchServiceHandlerDecorator.java:37) [modlauncher-8.0.6.jar:?]
	at cpw.mods.modlauncher.LaunchServiceHandler.launch(LaunchServiceHandler.java:54) [modlauncher-8.0.6.jar:?]
	at cpw.mods.modlauncher.LaunchServiceHandler.launch(LaunchServiceHandler.java:72) [modlauncher-8.0.6.jar:?]
	at cpw.mods.modlauncher.Launcher.run(Launcher.java:82) [modlauncher-8.0.6.jar:?]
	at cpw.mods.modlauncher.Launcher.main(Launcher.java:66) [modlauncher-8.0.6.jar:?]
```

**Comment by SDUBZ** [comment]
the later debug is without that mod
as i seen that before 
take note of the mods list of the last posted debug

**Comment by SDUBZ** [comment]
heres a lil snippet of the time frame
when minecolonies was present
[ 19:28:40.210] [Server thread/INFO] [net.minecraftforge.common.AdvancementLoadFix/]: Using new advancement loading for net.minecraft.advancements.PlayerAdvancements@4643204
[19Nov2020 19:39:09.130] [Server thread/INFO] [net.minecraft.server.MinecraftServer/]: Preparing start region for dimension minecraft:overworld]

**Comment by someaddons** [comment]
you likely just didnt set an appropriate amount of ram, and its taking ages because of that

**Comment by SDUBZ** [comment]
I had 8bgs allocated it should be more then enough @someaddons

**Comment by someaddons** [comment]
and its actually able to use ~12+ free gb of ram?

**Comment by SDUBZ** [comment]
yes i have 32 gbs total in my system

**Comment by Raycoms** [comment]
We don't do anything worldgen related. Do you have an ssd?

**Comment by SDUBZ** [comment]
yes i got a 250 gb ssd @Raycoms

**Comment by Raycoms** [comment]
Weird, this doesn't happen for us on any of our servers, not even on forgecraft

**Comment by SDUBZ** [comment]
thats why i posted the mod list to see fi maybe something you haven't got anywhere might be affecting it like a conflict that isnt outputting anything

**Comment by SDUBZ** [comment]
@Raycoms doesnt forgecraft push to the client from the server ? if so its not proper testing because it wouldnt load noramlly like most clients that dont run that config 🤔

**Comment by Raycoms** [comment]
Forgecraft just distributes the mods differently, the rest stays the same

**Comment by SDUBZ** [comment]
theres only one way to test the bug it would be to see whats diff between forgecraft and the pack im working on and see which ones affet minecolonies or it could be a single player bug too

**Comment by Raycoms** [comment]
Can you do an exclusion test and slowly remove mods to figure out which is causing this?
